### PR TITLE
SERVER-39195 Fix getUserDir to not solely rely on env var HOME

### PR DESCRIPTION
**Pull request by Yuxael**
Shell should not rely solely on env var HOME. It's very often not set at all when launching scripts e.g from cron or other apps. This causes shell to output a warning that it could not save .dbshell and thus making it very hard to parse output from shell as it would result in malformed JSON when retrieving data.
This change won't fix that kind of behaviour on MacOS as users settings are not kept in /etc/passwd so getpwuid() can't be used for this purpose.

### CODE REVIEW

**Comment by Schubes** [comment]
Hi @Yuxael,

Thank you for the pull request. So we can review these changes, would you please be sure to sign the [contributor agreement](https://www.mongodb.com/legal/contributor-agreement) and open a [SERVER ticket](https://jira.mongodb.org/browse/SERVER/) so we can track the fix?

Thanks again,
Kelsey

**Comment by Yuxael** [comment]
SERVER-39195
